include/nil/crypto3/zk/components/blueprint_linear_combination.hpp

```diff
@@ -110,13 +118,16 @@ namespace nil {
                     }
                 };
 
-                template<typename FieldType>
-                class blueprint_linear_combination_vector
-                    : private std::vector<blueprint_linear_combination<FieldType>> {
+                template<typename TArithmetizatio>
+                class blueprint_linear_combination_vector;
 
-                    typedef FieldType field_type;
-                    typedef typename field_type::value_type field_value_type;
-                    typedef std::vector<blueprint_linear_combination<field_type>> contents;
+                template<typename TBlueprintField>
+                class blueprint_linear_combination_vector<snark::r1cs_constraint_system<TBlueprintField>>
+                    : private std::vector<blueprint_linear_combination<snark::r1cs_constraint_system<TBlueprintField>>> {
```

Comment:
Using inheritance from `std::vector` should not be used because of the absence of proper memory management. Better to define strong typedef-alike data structure with explicit deallocation called.